flask_potion/schema.py

```diff
@@ -256,10 +256,13 @@ def convert(self, instance, update=False, pre_resolved_properties=None, patchabl
     def parse_request(self, request):
         if request.method in ('POST', 'PATCH', 'PUT', 'DELETE'):
             if self.fields and request.mimetype != 'application/json':
-                raise RequestMustBeJSON()
+                # Allow when all fields are optional
+                if not all(((i.default is not None) or i.nullable
+                                for i in self.fields.values())):
+                    raise RequestMustBeJSON()
 
         # TODO change to request.get_json(silent=False) to catch invalid JSON
-        data = request.get_json(silent=True)
+        data = request.get_json(silent=True) or {}
```

Comment:
Need this `or {}`, because there will be no JSON to parse and it'll return `None`.

Reply:
Instead, handle `None` below, i.e.

```
if data is None and self.is_optional:
    return {}
```

Reply:
I have to do `data = {}`. `return {}` causes a `TypeError` to be raised, when it enters the function. i.e. `TypeError: finalize() missing 1 required positional argument: 'total'`.

It appears to not pass the argument, when `return {}` is used.